### DISCUSSION TITLE
Can users translate text and apply it?

### DISCUSSION THREAD

**Discussion by Px2Px2** [aquasarus/Starfield-Interactive-Smart-Slate#26] [Q&A]
_* This article was written with a translator, so please excuse any inaccuracies. *_

It was only in hindsight that I found out about this useful and cool program.

I have no problem using it as it is, but it would be much easier if users could create their own translated versions. Is this possible, and if so, how would it be done? If you can show me how, I'll do it myself.
I guess I could just replace the French text in the localization folder, but I only have the flora and fauna text files here and I know I can enter them in my language in the program, so I'd like to translate the star system and resources used in the search into my language if possible.

Thank you for a great program. I'm excited to start recording with this awesome tool.

**Comment by aquasarus** [comment]
I'm glad you liked the app! Unfortunately a lot of the code in this app was pretty rushed. I wanted to make sure people could use it comfortably before I ran out of time. If you've browsed the code in here, you'll see that pretty much all the UI text is hardcoded. Here are your options:

If you only want to translate the names of the solar systems, planets, moons, and resources, these are stored in your database file ([see this page](https://github.com/aquasarus/Starfield-Interactive-Smart-Slate/wiki/User-Data)). All you need to do is download any free SQLite browser (I recommend [DB Browser for SQLite](https://sqlitebrowser.org/)), then do the following:
1. Close Starfield ISS.
2. IMPORTANT: make a backup copy of your `DataSlate.db` file in case you mess it up.
3. Open your `DataSlate.db` file with your SQLite browser.
4. Change the names in these tables: `Systems -> SystemName`, `CelestialBodies -> BodyName`, `Resources -> ResourceName`
5. Commit (i.e. click "Write Changes").
6. Reopen Starfield ISS.

If you want to translate the entire app, including the UI, you'll have to clone this repo and change it in the source code. You can download Visual Studio Code for free, and use it to compile the source code into your own version of the app.
